### DISCUSSION TITLE
Legend Added repeatedly when an actor was added again

### DISCUSSION THREAD

**Discussion by zhengliuX** [pyvista/pyvista#1928] [Q&A] [answered]
Hi, I am building an app using Pyvista. I encountered an weird problem. I created a dict, consists of some actors, like
```
actor = dict()
# create some mesh
mesh = ....
for label in labels:
    actor[label] = plotter.add_mesh(mesh[label])
    plotter.add_legend()
```
When I set a button to choose whether to remove an actor or add it again, the number of the legend of the re-added actor increased. When I did remove_actor, I did do plotter.remove_legend(), and the label of the actor is the same, so I'm not sure why this happened.

**Comment by banesullivan** [comment, marked as answer]
Try saving the `plotter.add_legend()` call until after your `for` loop. This is expected behavior as calling `plotter.add_legend()` multiple times would imply that multiple legends should be added

**Reply by zhengliuX** [reply]
Hi Thank you for your reply
 you mean like  ```lengend_actor[label] = plotter.add_legend()```

**Reply by banesullivan** [reply]
Something like:
```py
actor = dict()
# create some mesh
mesh = ....
for label in labels:
    actor[label] = plotter.add_mesh(mesh[label], label=label)
plotter.add_legend()
```

See https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_legend.html?highlight=add_legend#pyvista.Plotter.add_legend

**Reply by zhengliuX** [reply]
Hi, still added repeatly. 
This is what I did:
I havce some checkboxes with the name of some rois, and the rois is removed or added by clicking the checkboxes, also the legend related.
If I set some checkboxes checked, then uncheck them, then set them checked again.
In the BackgroundPlotter the should show uo, disappear, and show up again.
![image](https://user-images.githubusercontent.com/65115609/145017640-577e6e4a-2641-4583-a9c6-d0306a183992.png)

**Reply by zhengliuX** [reply]
OK, I think this might because that whenever  I added some rois again, I created a new mesh and add_mesh with a same label, if I add a judgement statement to check if this actor is created before to decide add_actor or add_mesh, this problem might disappear.

**Reply by zhengliuX** [reply]
OK, the error fixed!
